### ISSUE TITLE
/home test page feedback

### ISSUE THREAD

**Issue by roy7** [gcp/leela-zero-server#170]
The new front page for LZ is being tested here:

http://zero.sjeng.org/home

Please review and provide feedback!

**Comment by pcengine** [comment]
It looks really nice and neat. Great job!  One suggestion, is it possible to include estimated Elo under each network? Hopefully it's not too much work.  If it is, please just ignore my comments. 
Also, using real life estimated Elo scores (e.g. https://docs.google.com/spreadsheets/d/1QhYWYQvXBhI5eHJm45fvf97CaV2rRmgrATwVgBFGKoU/edit#gid=806323235 )  probably would be more useful for general go players.   I mean, if we see a network with an elo 3600, we can immediately know it's about the strongest human player's strength.  Some people might feel that the current LZ best's  11455 elo a bit confusing.  Just my two cents.

**Comment by l1t1** [comment]
suggest add a graph of winrate of each match, as following
![2018-06-14_160054](https://user-images.githubusercontent.com/23301704/41399017-c7eed510-6feb-11e8-902d-f33eef9463e7.png)

**Comment by 22nsuk** [comment]
I want to have a tab to download the raw sgf files and training files from the new home screen. By the way, the two links do not work right now.

**Comment by herazul** [comment]
Very cool, works well on mobile too.

Maybe i'm blind, but i don't see where we can watch some lastest self-play match ? The equivalent of this ? http://zero.sjeng.org/self-plays

And there is a padding/margin thing here that is triggering me, but i'm a crazy person ^^
![](https://i.imgur.com/HCje9I7.png)

**Comment by Manslay** [comment]
Looks very cool.

Is it possible to add winning statistics in the opening analysis section?

**Comment by Crazyeight101** [comment]
It looks like you're sacrificing functionality and ease of access to important or desirable information for some very slight improvements to look and feel.  If it were me, I would scale down the size of the graph, make the "recent matches" section much more compact and cleaner, add the recent self-play game and match game functionality back in somewhere, add the very useful text explanations on the current front page back in somewhere, and perhaps choose a different color scheme than black/gold/sky blue on a white background.  Maybe just find a simple image or pattern that captures some aspect or essence of the LZ project and use that for a banner, and then just leave the rest as black text on a white background, and maybe also some descriptive language about what the graph actually tells people.  

If you think about how people parse the information on the graph on the current page, most of them probably end up referring back and forth between the graph and the network list and test matches list to make good sense of the data, but with the new set up you're testing, they'd have to switch back and forth between 2 pages to do that, and the new network page is still less descriptive than the current one, so it would be more difficult to do in many ways.

As a general rule, the less pages you have to go through and the less clicks you need to get somewhere, the better.

The openings page is an interesting idea, but it looks a bit clunky and awkward to use.  Having to scroll down the page through a long list of what appear to be very simplistic joseki sequences with a hard to decipher scatter plot and some statistical data and no other functionality seems very nebulous.

From the standpoint of people already familiar with the project, the excessive use of space for each element and the dividing of important related elements on separate pages is just an irritating inconvenience.  But for new people who are coming to check it out for the first time, they're being presented with elements almost in isolation of one another that have no explanations about their relationships or importance, so it's like "oh, here's a graph", then they scroll down and "oh, here's a table of something, but I can't immediately tell what it is", and then they either spend several seconds to several minutes trying to figure it out, they stare at for a second it without bothering to try and understand it and continue searching for whatever it is that's important to them, or they just leave the website.


Honestly, there's very little that needs to be changed on the current one, except to clean up and organize the text a little bit with headers, maybe some simple stylistic features, and better placement, and a container for the self-play games list, and maybe an additional scroll bar for the list of networks and test matches so that you don't have to constantly scroll down a screen and lose the image of the graph in order to check information about the older networks.  Also, maybe some code to display the local time for each visitor instead of GMT+1.  And then, maybe, if it strikes your fancy, finding some simple art or pattern like I suggested above for a banner to add some character to the page and a navigation menu like you have on the new page for any new features you feel deserve a separate page, such as a bio on gcp and significant contributors, or a news feed with links to articles that reference or are about the project.

The most important thing that people need from the website is an easy overview of the status of the project, which the current page does very well.  My two cents.

**Comment by l1t1** [comment]
i also agree to keep current version, at least as a lite version for whom like it. the big and Deluxe version mainly add new features, such as opening lib, the duplicate features develop can be delay a little more time

**Comment by Crazyeight101** [comment]
Also, I just took a gander at the source, and wow. -.-

Might want to use some css and php to clean that up a bit. =P

**Comment by roy7** [comment]
@Cabu had a lot of suggestions over at https://github.com/gcp/leela-zero/issues/1555#issuecomment-397212788:

- Didn't found the link to download ALL the self play games and matches.
- Should the match-games pages be part of the relooking? (eg: [b69158ea vs 9e882e52](http://zero.sjeng.org/match-games/5b0d8221888bab42e3cde086))
- Should the viewmatch be part of the relooking too? (eg: [560e31...7d1ed9](http://zero.sjeng.org/viewmatch/560e31c34e3f386ddda6cb0d57f6579c06c6efdc6fbcea6ec9c31f65367d1ed9?viewer=wgo))
- On the network tab, there is no more ELO rating of the networks
- If we put the 10 last matches on the homepage, why not: 
  - the 5 lasts best networks also on the homepage?
  - ALL the matches on the matches pages?
- In the joseki list why not the [4-4 point low approach low extension, slide, 3-3](https://senseis.xmp.net/?44PointLowApproachLowExtensionSlide33)?
- There is a lot of unused data available in the HTML code but not used in the rendering. Could you lighten the page by getting rid of all the fluf (eg: _id, ip and description fields)?
- Could an [HTML checher](https://validator.w3.org/nu/?showsource=yes&doc=http%3A%2F%2Fzero.sjeng.org%2Fhome) run through it with less errors/warnings?

**Comment by Hersmunch** [comment]
The link on the front page to "Leela Zero 0.15 + AutoGTP v16. Update required." doesn't work. It goes to http://zero.sjeng.org/%22https://github.com/gcp/leela-zero/releases/%22

**Comment by carljohanr** [comment]
Openings section is a good idea (minigo pages have great information), but is quite confusing now. My best understanding is that the list of openings that are there now are hand-curated. I'd really enjoy something similar to that (e.g. how would Network X play out different fusekis with some standard settings). I realize this requires additional games or analysis, so may be out of scope for you.

**Comment by bvandenbon** [comment]
I love it. I think it's great. 

And I do think that the opening tag is a cool demo of "innovations" that LeeLa brings. I also like the scatter graph a lot, and I do think that its meaning is obvious. However, the horizontal axis of the graph is missleading. It runs all the way to 10 million even though LeeLa only played 7.8 million games so far. The maximum value of the horizontal axis should be 7.8 million, i.e. the number of played games. 

If you are looking for more patterns to add, here are some bot influences that I spotted in recent pro games:

![innovations](https://user-images.githubusercontent.com/20482760/41437101-f6268fe4-7023-11e8-8210-fb71782a47b7.png)
![overview](https://user-images.githubusercontent.com/20482760/41437009-c21c7c68-7023-11e8-8987-3626f08d8eb0.png)

As others pointed out, the most wonderful thing would be a more powerful opening library, e.g. an online pattern search website that has indexed those 7.8 million games (or the most recent ones). But really, please just stay focussed on what you're doing. :-) 'Cause you're doing a great job.

**Comment by Crazyeight101** [comment]
@bvandenbon The idea itself is interesting, but the implementation needs work.  Honestly, there's already a formula for it, more or less, which is an sgf tree on a single player, like eidogo, wgo, josekipedia, waltheri, and now Deepmind with the AZ teaching tool.  What none of those sites have that roy7 and whoever else may be working with him to build a newer site seem to be trying to build are analytics tools specific to the LZ project, like the scatter plot.

You may very well be able to understand the scatter plot, as would most people who are already familiar with the project, but with no clear explanation of the data you're going to lose anyone who's interested in the tools but isn't familiar with the project.

If the aim is to give the LZ project a face-lift and engender broader appeal, then it's an important point that needs to be addressed.  If it's just for the people already involved, I'm sure we've all gotten really comfortable with the current page and don't really care too much about L&F, although extra functionality, when it's done right, is always nice.

**Comment by roy7** [comment]
Just so you know the credit for this work largely goes to @rchs819 and @bood. 

@bvandenbon I love that Tang Weixing used the Leela Zero flower joseki!

**Comment by Crazyeight101** [comment]
@roy7 I see.  Thanks for the information. :D

**Comment by petgo3** [comment]
Download link does not work:
it's http://zero.sjeng.org/"https://github.com/gcp/leela-zero/releases/" instead of 
https://github.com/gcp/leela-zero/releases

**Comment by TFiFiE** [comment]
First of all, the use of the word "opening" for joseki is wrong, as many moves could already have been  played in the other quadrants. Also, the scatter plots should have a vertical line at the current number of training games so as not to falsely imply the joseki simply stopped being played past that point during non-existing games. It might also be interesting to have these statistics make a distinction between black and white being on one side or the other in these sequences.

**Comment by 3kakakhel** [comment]
### Foreword

I know I mostly lurk while providing training games, but I thought I might chime in anyway.  I understand some of these comments are more nitpicky than others. I apologize for the length. I have been working on this issue bit by bit. I hope the headings and bulleted lists increase the digestabilty. I also made a [mockup](https://i.imgur.com/Uzlt2qO.png) of the new homepage. It roughly incorporates my ideas and some others' ideas. Others have been contributing great ideas. I know it has some spacing issues. I am sure someone with better could resolve said issues.

:sweat_smile:

### Mockup

![Mockup](https://i.imgur.com/Uzlt2qO.png)

### Colors

* For the black menu bar, I might suggest `#666666` instead.
* For the yellow banner, I might suggest `#ffe6cc` instead.
* For the blue banner, I might suggest `#cce6ff` instead.
* For links, I might suggest `#5b9bd5` instead.
* For the `Failed` SPRT tag the matches column, I might suggest `#ffcce6` for the background instead.
* For the `Pass` SPRT tag in the matches column, I might suggest `#e6ffcc` for the background instead.
* The SPRT tags could also include a `Promoted` tag for matches that promote a network using `#cce6ff`.
* The SPRT tags could also include a `Test` tag for test matches using `#e6e6e6`.
* ![#666666](https://placehold.it/15/666666/000000?text=+) #666666
* ![#e6e6e6](https://placehold.it/15/e6e6e6/000000?text=+) #e6e6e6
* ![#ffe6cc](https://placehold.it/15/ffe6cc/000000?text=+) #ffe6cc
* ![#cce6ff](https://placehold.it/15/cce6ff/000000?text=+) #cce6ff
* ![#5b9bd5](https://placehold.it/15/5b9bd5/000000?text=+) #5b9bd5
* ![#ffcce6](https://placehold.it/15/ffcce6/000000?text=+) #ffcce6
* ![#e6ffcc](https://placehold.it/15/e6ffcc/000000?text=+) #e6ffcc

### FAQ

* What is already in the FAQ.md page.
* Helpful info already in the old layout.
* Rollouts vs playouts vs visits.
* Required version numbers.
* Other repeated questions.

### SGFs

* I agree with Cabu and 22nsuk, there should be a link to download all the self play games and matches and trainging data. Maybe add a page for that and link in the menubar at the top.
* SGFs for self-play games.
* SGFs for match games.
* Training data.

### Self-Plays

* I agree with herazul, there should be a page for the self-play games.
* The table on this page should include all the info from the old layout but with the new layout's formatting.

### Matches

* I agree with Cabu, that the VS match links should be part of the new look too.
* I would suggest keeping the win rate, winner, score, move count, duration, sgf, and client columns from the old layout.

### Networks

* I agree with Cabu and pcengine, the elo rating should be added back for the networks. Not sure how to include other elo ratings though. The CGOS rating might still not be familiar to the general public. The most interesting would be a corresponding elo placement using the [goratings](http://goratings.com) elo scale, what I call pro elo. I am not sure how to accomplish such a task.
* I would suggest separating the cumulative games played and training steps into their own columns in the new layout.

### Joseki

* I would suggest 'Total Training Games' for the x-axis label instead.
* People may be curious about other starting points such as 10-10, 5-4, 5-10.
* Did the 4-4 Point joseki Cabu mentions really not appear often enough for inclusion?
* The plots on the 4-4 Point josekis are much smaller than the plots for the other josekis.
* I would suggest a highlight color when mousing over the 4-4 Point, 4-3 Point, and 3-3 Point headings.
* The bulleted list in the tooltips in the scatter plots in the openings section do not fit inside the tooltip.
* I would suggest giving said tooltips a light yellow background, maybe with some transparency.
* I agree with Manslay, winning statistics on the josekis by Leela Zero would be interesting.
* I like the tooltips when mousing over the scatter plots in the openings section. Their presence and information is a positive addition.

### Personal Preferences

* I prefer the equal width of the numbers of the old layout, despite not being a fixed width font.
* I prefer to be able to click on a table heading to toggle sorting by that column like in the old layout.
* I prefer the datetime display of the old layout: `yyyy-mm-dd hh:mm` and have it be consistent throughout the website.

### Commentary

* I like the scrape-ability of the old layout.
* I like the copy-paste-ability of the old layout.
* I like the idea to modernize the look and feel.
* I like that each network gets its own profile page.
* I like that the datetime gets converted to my timezone.
* I like that the current row gets highlighted on mouse over.
* I agree with Crazyeight101, the links in the new layout are not obvious.
* The page tries to load as the desktop version on my phone. Also no switching happens if I simply reduce the width of my browser. WIth the mockup, the mobile version would mainly be different in that the menubar at the top would transform into a hamberger menu with the Leela Zero label showing.
* I would suggest having thousands separators for all the large numbers across the website. I would prefer using a space, but not required.

**Comment by l1t1** [comment]
![2018-06-20_101808](https://user-images.githubusercontent.com/23301704/41633768-fe773d4e-7472-11e8-9b10-a11051212825.png)
it shows old message(now lz150, it shows lz148), and the result of top 2 lines is error

**Comment by bood** [comment]
@l1t1 Fixed in #180

**Comment by l1t1** [comment]
recent matchs doesn't show on chrome
![2018-06-23_144338](https://user-images.githubusercontent.com/23301704/41806625-da583248-76b0-11e8-9dec-1f2a55b6645b.png)
![2018-06-23_144556](https://user-images.githubusercontent.com/23301704/41806637-0cb71376-76b1-11e8-9122-ec84800aad74.png)

**Comment by l1t1** [comment]
@bood @l1t1 Fixed in #180
latest  match
http://zero.sjeng.org/ shows 314 games
http://zero.sjeng.org/match-games/5b30226d9ecc2573bd5720ae also shows 314
but http://zero.sjeng.org/home shows 315
![2018-06-25_103944](https://user-images.githubusercontent.com/23301704/41827853-6100d76e-7864-11e8-8748-a8fe7d664145.png)
![2018-06-25_104019](https://user-images.githubusercontent.com/23301704/41827859-655b596a-7864-11e8-8138-6faf37ebc8f7.png)
![2018-06-25_104100](https://user-images.githubusercontent.com/23301704/41827860-6897a250-7864-11e8-8df7-7a97bf64d2d8.png)

**Comment by gjm11** [comment]
On the "old" home page, hovering over a network's short name in the match list gives more information than it does on the "new" home page. E.g., for the ELF network it tells you it's ELF, says which version of the ELF weights, etc.; for regular LZ networks it tells you the network architecture. It would be nice to have that back.

The match list takes up a lot more space on the "new" page than the old. Perhaps some of the information currently in hover text could be un-hidden. I'm thinking in particular of the test / regular match information, without which the table is kinda misleading since when there's just been a match against something strong like ELF it looks as if the LZ networks have suddenly made huge progress :-).

**Comment by gjm11** [comment]
I think /home would be friendlier to newcomers and more useful to everyone if (1) it displayed at least a little information about *what the heck it is* and (2) it had one line of info saying what the latest network is, perhaps also when it was last updated, and with a link to download it.
For (1) I'm thinking something like "Leela Zero is a strong go-playing program trained by self-play on the computers of thousands of volunteers. For more information, see the README" with "README" being a link. If the FAQ were more approachable, there could be a link to that too. Might mention AlphaGo Zero too, might not. For (2), something like "Latest network is LZ153, 15 blocks x 256 channels, from 2018-06-29 at 18:41." with "LZ153" being a link to the actual weights file.

**Comment by l1t1** [comment]
sugguest add a list of  link to strongest weight of each size.
eg.
https://github.com/gcp/leela-zero/files/2162485/best_5b.txt.gz at https://github.com/gcp/leela-zero/issues/1557#issuecomment-402580126

**Comment by kennychenfs** [comment]
How often do you update Joseki?

**Comment by kennychenfs** [comment]
The number of Joseki appearances should be wider, as some of the Josekis appear intensive.

**Comment by l1t1** [comment]
could you add a graph like this ( https://github.com/gcp/leela-zero/issues/2058#issuecomment-443094677 ) for each match?

**Comment by iopq** [comment]
I can't zoom out in mobile, so I can't see anything on my phone

**Comment by kennychenfs** [comment]
How do you update joseki?  
I think that is too long time ago.

**Comment by roy7** [comment]
I don't actually know but I think @sethtroisi was looking at it.

**Comment by PJTraill** [comment]
Could you not include commas in all those long numbers? It makes them much easier for people and not much harder for programmes to read. (The site https://zero.sjeng.org/ is down (502), so I cannot check out more right now.)

**Comment by kennychenfs** [comment]
@sethtroisi 
Please update the joseki, which has been 541 days ago!
It must have many many countless changes of josekis, please update it.
